test/spec/lateralus.js

```diff
@@ -359,6 +359,36 @@ define([
               assert.deepEqual(collectedTest, [5]);
             });
           });
+
+          describe('Returned array contains no undefined values', function () {
+            var App = getLateralusApp();
+
+            _.extend(App.prototype, {
+              provide: {
+                test: function () {
+                  return;
+                }
+              }
+            });
+
+            var ExtendedComponent = Lateralus.Component.extend({
+              name: 'extended'
+              ,provide: {
+                test: function () {
+                  return 1;
+                }
+              }
+            });
```

Comment:
Similarly, I would include a test that has at least one falsy value in the array. I was going to suggest using `_.compact`, before I realized that was too aggressive.
A test case that catches some other bright spark making that change would be a good idea.